quickwit-janitor/src/actors/garbage_collector.rs

```diff
@@ -115,30 +110,71 @@ impl Handler<Loop> for GarbageCollector {
         info!("garbage-collect-operation");
         self.counters.num_passes += 1;
 
-        let deleted_file_entries = run_garbage_collect(
-            &self.pipeline_id.index_id,
-            self.split_store.clone(),
-            self.metastore.clone(),
-            STAGED_GRACE_PERIOD,
-            DELETION_GRACE_PERIOD,
-            false,
-            Some(ctx),
-        )
-        .await?;
-
-        if !deleted_file_entries.is_empty() {
-            let deleted_files: HashSet<&str> = deleted_file_entries
-                .iter()
-                .map(|deleted_entry| deleted_entry.file_name.as_str())
-                .collect();
-            info!(deleted_files=?deleted_files, "gc-delete");
-
-            self.counters.num_deleted_files += deleted_file_entries.len();
-            self.counters.num_deleted_bytes += deleted_file_entries
-                .iter()
-                .map(|entry| entry.file_size_in_bytes as usize)
-                .sum::<usize>();
+        let index_metadatas = self
+            .metastore
+            .list_indexes_metadatas()
+            .await
+            .context("Failed to list indexes.")?;
```

Comment:
Since the GC actor is running unsupervised, this will crash and stop the GC on the janitor service.  Let's properly log the error and return early.

The GC actor is a glorified for-loop, so I agree with the unsupervised strategy but let's be more resilient to errors.